src/main/java/se/sundsvall/digitalregisteredletter/api/EligibilityResource.java

```diff
@@ -0,0 +1,46 @@
+package se.sundsvall.digitalregisteredletter.api;
+
+import static org.springframework.http.MediaType.APPLICATION_PROBLEM_JSON_VALUE;
+import static org.springframework.http.ResponseEntity.ok;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import java.util.List;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.zalando.problem.Problem;
+import org.zalando.problem.violations.ConstraintViolationProblem;
+import se.sundsvall.digitalregisteredletter.api.model.EligibilityRequest;
+import se.sundsvall.digitalregisteredletter.service.EligibilityService;
+
+@RestController
+@Validated
+@RequestMapping("/{municipalityId}/eligibility")
+@Tag(name = "Eligibility Resource", description = "Check eligibility for sending digital registered letters")
+@ApiResponse(responseCode = "200", description = "Successful operation", useReturnTypeSchema = true)
+@ApiResponse(responseCode = "400", description = "Bad request", content = @Content(mediaType = APPLICATION_PROBLEM_JSON_VALUE, schema = @Schema(oneOf = {
+	Problem.class, ConstraintViolationProblem.class
+})))
+@ApiResponse(responseCode = "500", description = "Internal Server error", content = @Content(mediaType = APPLICATION_PROBLEM_JSON_VALUE, schema = @Schema(implementation = Problem.class)))
+class EligibilityResource {
+
+	private final EligibilityService eligibilityService;
+
+	EligibilityResource(final EligibilityService eligibilityService) {
+		this.eligibilityService = eligibilityService;
+	}
+
+	@GetMapping("/kivra")
+	@Operation(summary = "Check if the given partyIds are eligible for receiving digital registered letters with Kivra", description = "Returns a list of party IDs that are eligible for Kivra based on the provided municipality ID and party IDs")
+	ResponseEntity<List<String>> checkKivraEligibility(@PathVariable("municipalityId") final String municipalityId, final EligibilityRequest request) {
```

Comment:
Minor:
Name on `@PathVariable` is not recommended when the name is equal to the parameter (as in the code above).

Reply:
recommended by whom?

Reply:
By Spring? (or at least the Spring tools suite):

<img width="598" height="227" alt="Skärmbild 2025-08-20 084140" src="https://github.com/user-attachments/assets/35eb4f95-b22e-4dc1-b552-636a9c49c1f2" />